docs/standard/microservices-architecture/microservice-ddd-cqrs-patterns/nosql-database-persistence-infrastructure.md

```diff
@@ -119,16 +116,219 @@ You can see that the way you work with your domain model can be similar to the w
+    {
+        get
+        {
+            return _database.GetCollection<Locations>("Locations");
+        }
+    }       
+}
+```
+
+#### Retrieve the data
+
+Then, from any C# code, like Web API controllers or custom Repositories, you can write similar code to the following when querying through the MongoDB API, being the `_context` object an instance of the previous `LocationsContext` class.
+
+```csharp
+public async Task<Locations> GetAsync(int locationId)
+{
+    var filter = Builders<Locations>.Filter.Eq("LocationId", locationId);
+    return await _context.Locations
+                            .Find(filter)
+                            .FirstOrDefaultAsync();
+}
+```
+
+#### Use an env-var in the docker-compose.override.yml file for the MongoDB connection string
+
+When creating a MongoClient object, the needed parameter is precisely the ConnectionString pointing, in the case of eShopOnContainers, either to a local MongoDB Docker container or to the “production” Azure Cosmos DB database. That connection string comes from the envitronment variables defined at the docker-compose.override.yml files when deploying with docker-compose or Visual Studio, as in the following yml code.
+
+```yml
+# docker-compose.override.yml
+version: '3'
+services:
+  # Other services
+  locations.api:
+    environment:
+      # Other settings
+      - ConnectionString=${ESHOP_AZURE_COSMOSDB:-mongodb://nosql.data}
+
+```
+
+The `ConnectionString` environment vartiable is defined so if the `ESHOP_AZURE_COSMOSDB` global variable is defined at the `.env` file with the Azure Cosmos DB connection string, it will it to access the Azure Cosmos DB database in the cloud. 
+
+Here you can see the `.env` file with the Azure Cosmos DB connection string gloval environment variable, used at eShopOnContainers as a convenient approach.
+
+```yml
+# .env file, in eShopOnContainers root folder
+# Other Docker environment variables
+
+ESHOP_EXTERNAL_DNS_NAME_OR_IP=localhost
+ESHOP_PROD_EXTERNAL_DNS_NAME_OR_IP=<YourDockerHostIP>
+
+#ESHOP_AZURE_COSMOSDB=<YourAzureCosmosDBConnData>
+
+#Other environment variables for additional Azure infrastructure assets
+#ESHOP_AZURE_REDIS_BASKET_DB=<YourAzureRedisBasketInfo>
+#ESHOP_AZURE_STORAGE_CATALOG_URL=<YourAzureStorage_Catalog_BLOB_URL>
+#ESHOP_AZURE_SERVICE_BUS=<YourAzureServiceBusInfo>
+```
+
+You should uncomment the ESHOP_AZURE_COSMOSDB line and updating it with your real Azure Cosmos DB connection string obtained from the Azure potal as explained in this tutorial named [Connect a MongoDB application to Azure Cosmos DB](https://docs.microsoft.com/en-us/azure/cosmos-db/connect-mongodb-account).
+
+If the `ESHOP_AZURE_COSMOSDB` global variable is empty, meaning that it is commented at the `.env` file, then it will use a by default MongoDB connection string pointing to the local MongoDB container deployed in eShopOnContainers which is named `nosql.data` and was defined at the docker-compose file, as shown in the following .yml code. 
```

Comment:
commented at --> comment out in 
it will use a by default --> the container uses a default 

If the `ESHOP_AZURE_COSMOSDB` global variable is empty, meaning that it is commented at the `.env` file, then it will use a by default MongoDB connection string pointing to the local MongoDB container deployed in eShopOnContainers which is named `nosql.data` and was defined at the docker-compose file, --> This sentence is extremely difficult to parse. Rewrite it for clarity.

There is no following .yml code

Reply:
👍 - Changed.